app/controllers/api_controller.rb

```diff
@@ -156,44 +156,30 @@ def initialize
   #
   include_concern 'Initializer'
 
+  before_action :parse_api_request, :log_api_request, :validate_api_request
+  after_action :log_api_response
+
   def self.attr_type_hash(type)
     instance_variable_get("@attr_#{type}") || {}
   end
 
   def redirect_api_request(method)
     target_method = "#{method}_#{@req.collection || "entrypoint"}"
-    if respond_to?(target_method)
-      send(target_method)
-      return true
-    end
+    return send(target_method) if respond_to?(target_method)
     target_method = "#{method}_generic"
-    if respond_to?(target_method)
-      send(target_method)
-      return true
-    end
-    false
-  end
-
-  #
-  # REST APIs Handler and API Entrypoints
-  #
-  def api_request_handler(expected_method)
-    parse_api_request
-    log_api_request
-    validate_api_request
-    api_error_type(:not_found, "Unknown resource specified") unless redirect_api_request(expected_method)
-    log_api_response
```

Comment:
can you test the API with this change via a browser (with no auth, let the browser popup the authentication) if you haven't ?  Thanks.  make sure the request is still properly logged in api.log.